### PR TITLE
hdf5 - upgrade for conan v2, simplify szip import

### PR DESCRIPTION
**Pull request by paulharris**
HDF5's cmake files have problems importing our szip and friend, so simplified that bit.

### CODE REVIEW

File: recipes/hdf5/all/conanfile.py [hunk 1]
```diff
@@ -39,123 +43,119 @@ class Hdf5Conan(ConanFile):
...
             del self.options.fPIC
 
+    def validate(self):
+        if not can_run(self):
+            # While building it runs some executables like H5detect
+            raise ConanInvalidConfiguration("Current recipe doesn't support cross-building (yet)")
+        if self.info.options.parallel:
+            if self.info.options.enable_cxx:
+                raise ConanInvalidConfiguration("Parallel and C++ options are mutually exclusive")
+            if self.info.options.get_safe("threadsafe"): # FIXME why can't I define the default valid as False?
```
Comment (prince-chrismc) [inline recipes/hdf5/all/conanfile.py:60]: What exactly is the issue?

**Review by prince-chrismc — DISMISSED** [review]

File: recipes/hdf5/all/conanfile.py (hunk 1 above)
Comment (paulharris) [reply recipes/hdf5/all/conanfile.py:60]: This is apparently some built-in limitation with HDF5.
Parallel == MPI so its not just about enabling threads.
I think only Linux and OpenMPI is possible at the moment too.

**Comment by paulharris** [comment]
Clang-11 with libc++ 
why does this happen?
`H5StrType.cpp:276: undefined reference to `std::allocator<char>::allocator()'`
I'm sure I've seen this before ... something to do with C++11 ABI...

Only an old version, 1.12.0, has this problem.
I can't drop the version as an old version of NetCDF needs it.

**Comment by paulharris** [comment]
@prince-chrismc I think this recipe is done

**Review by SSE4 — APPROVED** [review]

**Pull request merged**